net/pfSense-pkg-WireGuard/files/usr/local/www/wg/status_wireguard.php

```diff
@@ -1,129 +1,240 @@
+ */
+
+##|+PRIV
+##|*IDENT=page-status-wireguard
+##|*NAME=Status: WireGuard
+##|*DESCR=Allow access to the 'Status: WireGuard' page.
+##|*MATCH=status_wireguard.php*
+##|-PRIV
+
+// pfSense includes
+require_once('guiconfig.inc');
+require_once('util.inc');
+
+// WireGuard includes
+require_once('wireguard/wg.inc');
+require_once('wireguard/wg_guiconfig.inc');
+
+// Grab the latest info
+wg_globals();
+
+global $wgg;
+
+$shortcut_section = "wireguard";
+
+$pgtitle = array(gettext("Status"), gettext("WireGuard"));
+$pglinks = array("", "@self");
+
+$tab_array = array();
+$tab_array[] = array(gettext("Tunnels"), false, "/wg/vpn_wg_tunnels.php");
+$tab_array[] = array(gettext("Peers"), false, "/wg/vpn_wg_peers.php");
+$tab_array[] = array(gettext("Settings"), false, "/wg/vpn_wg_settings.php");
+$tab_array[] = array(gettext("Status"), true, "/wg/status_wireguard.php");
+
+include("head.inc");
+
+wg_display_service_warning(false);
+
+display_top_tabs($tab_array);
+
+$a_devices = wg_status();
+
+if (!empty($a_devices)):
+
+?>
+
+<div class="panel panel-default">
+	<div class="panel-heading">
+		<h2 class="panel-title"><?=gettext('WireGuard Status')?></h2>
+	</div>
+	<div class="table-responsive panel-body">
+		<table class="table table-hover table-striped table-condensed" style="overflow-x: 'visible'">
+			<thead>
+				<th><?=gettext('Tunnel')?></th>
+				<th><?=gettext('Description')?></th>
+				<th><?=gettext('# Peers')?></th>
+				<th><?=gettext('Public Key')?></th>
+				<th><?=gettext('Address / Assignment')?></th>
+				<th><?=gettext('MTU')?></th>
+				<th><?=gettext('Listen Port')?></th>
+				<th><?=gettext('RX')?></th>
+				<th><?=gettext('TX')?></th>
+			</thead>
+<?php
+	foreach ($a_devices as $device_name => $device):
+		wg_
+?>
+			<tbody>	
+				<tr class="tunnel-entry">
+					<td>
+						<?=wg_interface_status_icon($device['status'])?>
+						<a href="vpn_wg_tunnels_edit.php?tun=<?=htmlspecialchars($device_name)?>"><?=htmlspecialchars($device_name)?>
+					</td>
+					<td><?=htmlspecialchars(wg_truncate_pretty($device['config']['descr'], 16))?></td>
+					<td><?=count($device['peers'])?></td>
+					<td><?=htmlspecialchars(wg_truncate_pretty($device['public_key'], 16))?></td>
+					<td><?=wg_generate_tunnel_address_popup_link($device_name)?></td>
+					<td><?=htmlspecialchars($device['mtu'])?></td>
+					<td><?=htmlspecialchars($device['listen_port'])?></td>
+					<td><?=htmlspecialchars(format_bytes($device['transfer_rx']))?></td>
+					<td><?=htmlspecialchars(format_bytes($device['transfer_tx']))?></td>
+				</tr>
+				<tr class="peer-entries">
+	
+<?php
+		if ($device['status'] == 'up'):
+			
+			if (count($device['peers']) > 0):
+?>
+					<td colspan="9">
+						<table class="table table-hover table-condensed">
+							<thead>
+								<th><?=gettext('Peer')?></th>
+								<th><?=gettext('Latest Handshake')?></th>
+								<th><?=gettext('Public Key')?></th>
+								<th><?=wg_format_endpoint(true)?></th>

```

Comment:
This one lost the `htmlspecialchars()` in one of your last commits

Reply:
fixed